osutil/resolve_path_test.go

```diff
@@ -0,0 +1,173 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil_test
+
+import (
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type resolvePathSuite struct{}
+
+var _ = Suite(&resolvePathSuite{})
+
+func (s *resolvePathSuite) TestSimple(c *C) {
```

Comment:
I wonder if each of the tests should also test the non-escape variant? Something like:
```go
diff --git a/osutil/resolve_path_test.go b/osutil/resolve_path_test.go
index c6d6d615d4..b9c0e91ce3 100644
--- a/osutil/resolve_path_test.go
+++ b/osutil/resolve_path_test.go
@@ -40,6 +40,10 @@ func (s *resolvePathSuite) TestSimple(c *C) {
        resolved, err := osutil.ResolvePathInSysroot(sysroot, "/foo/bar")
        c.Assert(err, IsNil)
        c.Assert(resolved, Equals, "/foo/bar")
+
+       resolved, err = osutil.ResolvePathNoEscape(sysroot, "/foo/bar")
+       c.Assert(err, IsNil)
+       c.Assert(resolved, Equals, "/foo/bar")
 }
 
 func (s *resolvePathSuite) TestSimpleRelative(c *C) {
@@ -50,6 +54,10 @@ func (s *resolvePathSuite) TestSimpleRelative(c *C) {
        resolved, err := osutil.ResolvePathInSysroot(sysroot, "foo/bar")
        c.Assert(err, IsNil)
        c.Assert(resolved, Equals, "/foo/bar")
+
+       resolved, err = osutil.ResolvePathNoEscape(sysroot, "foo/bar")
+       c.Assert(err, IsNil)
+       c.Assert(resolved, Equals, "/foo/bar")
 }
 
 func (s *resolvePathSuite) TestDot(c *C) {
@@ -60,6 +68,10 @@ func (s *resolvePathSuite) TestDot(c *C) {
        resolved, err := osutil.ResolvePathInSysroot(sysroot, "/./foo/./bar/.")
        c.Assert(err, IsNil)
        c.Assert(resolved, Equals, "/foo/bar")
+
+       resolved, err = osutil.ResolvePathNoEscape(sysroot, "/./foo/./bar/.")
+       c.Assert(err, IsNil)
+       c.Assert(resolved, Equals, "/foo/bar")
 }
 
 func (s *resolvePathSuite) TestEmpty(c *C) {
@@ -80,6 +92,9 @@ func (s *resolvePathSuite) TestDotDot(c *C) {
        resolved, err := osutil.ResolvePathInSysroot(sysroot, "../../../../foo/bar")
        c.Assert(err, IsNil)
        c.Assert(resolved, Equals, "/foo/bar")
+
+       _, err = osutil.ResolvePathNoEscape(sysroot, "../../../../foo/bar")
+       c.Assert(err, ErrorMatches, "invalid escaping path")
 }
 
 func (s *resolvePathSuite) TestDotDotInSymlink(c *C) {
```
etc

I know right now it's the same underlying implementation so it's a bit silly but it would "protect" against future refactors that might change the implementation. Right now AFAICT only the negative cases are explicitly tested for ResolvePathNoEscape. Or maybe a table driven test like:
```go
func (s *resolvePathSuite) TestHappy(c *C) {
	for _, tc := range []struct {
		path     string
		expected string
	}{
		// simple
		{"/foo/bar", "/foo/bar"},
		// relative path
		{"foo/bar", "/foo/bar"},
		// dot
		{"/./foo/./bar/.", "/foo/bar"},
		// empty
		{"//foo/////bar//", "/foo/bar"},
	} {
		sysroot := c.MkDir()
		err := os.MkdirAll(filepath.Join(sysroot, "foo/bar"), 0700)
		c.Assert(err, IsNil)

		resolved, err := osutil.ResolvePathInSysroot(sysroot, tc.path)
		c.Assert(err, IsNil)
		c.Assert(resolved, Equals, tc.expected)

		resolved, err = osutil.ResolvePathNoEscape(sysroot, tc.path)
		c.Assert(err, IsNil)
		c.Assert(resolved, Equals, tc.expected)
	}
}
```

Reply:
I have parameterized those common tests and added tests for the `NoEscape` version. So that we get errors for exactly the tests that are broken rather having to guess from variables or line number.